ambry-store/src/main/java/com.github.ambry.store/BlobStore.java

```diff
@@ -76,6 +77,7 @@
   private BlobStoreStats blobStoreStats;
   private boolean started;
   private FileLock fileLock;
+  private ReplicaState currentState;
```

Comment:
this should probably be volatile since it will be set from a different thread than where it is read

Reply:
fair point, will fix